File: mo-auth/src/main/java/ink/ckx/mo/auth/authentication/captcha/CaptchaGrantAuthenticationToken.java
package ink.ckx.mo.auth.authentication.captcha;

import ink.ckx.mo.auth.authentication.base.BaseAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.oauth2.core.AuthorizationGrantType;

import java.util.Map;
import java.util.Set;

/**
 * @author chenkaixin
 * @description
 * @since 2023/12/04
 */
public class CaptchaGrantAuthenticationToken extends BaseAuthenticationToken {

    public CaptchaGrantAuthenticationToken(AuthorizationGrantType authorizationGrantType, Authentication clientPrincipal, Set<String> scopes, Map<String, Object> additionalParameters) {
        super(authorizationGrantType, clientPrincipal, scopes, additionalParameters);
    }
}

File: mo-auth/src/main/kotlin/ink/ckx/mo/auth/authentication/captcha/CaptchaGrantAuthenticationToken.kt
package ink.ckx.mo.auth.authentication.captcha

import ink.ckx.mo.auth.authentication.base.BaseAuthenticationToken
import org.springframework.security.core.Authentication
import org.springframework.security.oauth2.core.AuthorizationGrantType

/**
 * @author chenkaixin
 * @description
 * @since 2023/12/04
 */
class CaptchaGrantAuthenticationToken(
    authorizationGrantType: AuthorizationGrantType,
    clientPrincipal: Authentication,
    scopes: Set<String>,
    additionalParameters: Map<String, Any>
) : BaseAuthenticationToken(authorizationGrantType, clientPrincipal, scopes, additionalParameters)